aten/src/ATen/native/Loss.cpp

```diff
@@ -340,6 +342,49 @@ Tensor smooth_l1_loss_backward(const Tensor& grad_output, const Tensor& input, c
   return at::smooth_l1_loss_backward_out(grad_input, grad_output, input, target, reduction, beta);
 }
 
+Tensor huber_loss(const Tensor& input, const Tensor& target, const int64_t reduction, double beta) {
+  TORCH_CHECK(beta > 0, "huber_loss does not support non-positive values for beta.")
+  Tensor loss;
+  auto iter = TensorIterator::binary_op(loss, input, target);
+  huber_stub(iter.device_type(), iter, beta);
```

Comment:
Good question! I did this to follow what the mse and smooth l1 losses do, and can only imagine that they did this for weird premature optimization reasons. Since I don't see any real reason not to, I'm going to do as you suggested here :)

Reply:
After trying to rewrite this nicely for a bit, I realized a couple things:

1. The current `huber_loss_out`, based on the out variant of other losses (e.g. smooth l1, mse, etc.), does not reduce correctly. See [50382](https://github.com/pytorch/pytorch/issues/50382). I fixed the behavior for huber loss.
2. With the fix in place, calling `huber_loss_out` from `huber_loss` results in an unnecessary copy. I think this will be clear in the new implementation.

So for max performance, there's some duplication :( It's possible I've overlooked the perfect way to do this that results in performant versions of both `huber_loss` and `huber_loss_out`, so if you see it, please point it out!
* If we don't care about `huber_loss_out` performance, we can call `huber_loss` from it.
* Alternatively, we could remove the out variant entirely. In fact, I am curious what out variants are used for?